### PR TITLE
Bug 1184482 - play silence ringer during the vibration mode

### PR DESCRIPTION
**Pull request by alastor0325**
Play the silence ringer to participate the audio channel competing, when the phone is in the vibration mode.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6aa9ba4e9801ee6865f2ce0246fee3a4cc1e5b19

File: apps/system/js/dialer_agent.js [hunk 1]
```diff
@@ -192,6 +185,16 @@
     });
   };
 
+  DialerAgent.prototype._playRing = function da_playRing () {
+    // Notice that, even we are in the vibration mode, we would still play the
+    // silence ringer. That is because when the incoming call is coming, other
+    // playing sound should be paused. Therefore, we need to a silence ringer
+    // to participate the audio competing.
```
Comment (gabrielesvelto) [inline apps/system/js/dialer_agent.js]: nit: to participate the audio competing -> to compete for the audio output

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6b8a6ca436451a6e9a0603c0fc3b18c44c60a1dd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1790114b1fd9543a80dcd20c2532b7076f58b566

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=375a92a350a2ae53398cbd5096588524e735adad

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=786634e690aa844da01aaf6cd260c34b688b9548

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=781dee5aa8f539e8b49fbdcb28f120e8848433ef

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=556d3016e2093904ce2539d6bfb17db54edcf52b

File: apps/system/test/unit/dialer_agent_test.js [hunk 2]
```diff
@@ -219,8 +219,8 @@ suite('system/DialerAgent', function() {
         callschanged();
       });
 
-      test('it should not play the ringtone', function() {
-        assert.isTrue(mockAudio.play.notCalled);
+      test('it should play the silent ringtone', function() {
+        assert.isTrue(mockAudio.play.calledOnce);
```
Comment (gabrielesvelto) [inline apps/system/test/unit/dialer_agent_test.js]: nit: You can use the new format for sinon assertions which is usually more terse
```javascript
sinon.assert.calledOnce(mockAudio.play)
```

File: apps/system/test/unit/dialer_agent_test.js [hunk 3]
```diff
@@ -270,12 +270,34 @@ suite('system/DialerAgent', function() {
...
-    suite('even if the vibration is enabled', function() {
+    suite('if the vibration is enabled', function() {
       setup(function() {
         MockSettingsListener.mTriggerCallback('vibration.enabled', true);
+        callschanged();
       });
 
-      test('it should not vibrate', function() {
+      test('it should start vibrating', function() {
+        assert.isTrue(vibrateSpy.calledWith([200]));
```
Comment (gabrielesvelto) [inline apps/system/test/unit/dialer_agent_test.js]: nit: Likewise
```javascript
sinon.assert.calledWith(vibrateSpy, [200]);
```

File: apps/system/test/unit/dialer_agent_test.js [hunk 4]
```diff
@@ -270,12 +270,34 @@ suite('system/DialerAgent', function() {
...
+      });
+
+      test('it should vibrate every 600ms', function() {
+        this.sinon.clock.tick(600);
+        assert.isTrue(vibrateSpy.calledTwice);
```
Comment (gabrielesvelto) [inline apps/system/test/unit/dialer_agent_test.js]: nit: Likewise

File: apps/system/test/unit/dialer_agent_test.js [hunk 5]
```diff
@@ -270,12 +270,34 @@ suite('system/DialerAgent', function() {
...
+        this.sinon.clock.tick(600);
+        assert.isTrue(vibrateSpy.calledThrice);
```
Comment (gabrielesvelto) [inline apps/system/test/unit/dialer_agent_test.js]: nit: Likewise

File: apps/system/test/unit/dialer_agent_test.js [hunk 6]
```diff
@@ -270,12 +270,34 @@ suite('system/DialerAgent', function() {
...
+      });
+
+      test('it should stop when the call state changes', function() {
+        secondCall.addEventListener.yield();
+        vibrateSpy.reset();
+        this.sinon.clock.tick(600);
+        assert.isTrue(vibrateSpy.notCalled);
```
Comment (gabrielesvelto) [inline apps/system/test/unit/dialer_agent_test.js]: nit: Likewise

File: apps/system/test/unit/dialer_agent_test.js [hunk 7]
```diff
@@ -285,14 +307,10 @@ suite('system/DialerAgent', function() {
         MockSettingsListener.mTriggerCallback('audio.volume.notification', 7);
       });
 
-      test('it should not play', function() {
-        assert.isTrue(mockAudio.play.notCalled);
+      test('it should play the silent ringtone', function() {
+        assert.isTrue(mockAudio.play.called);
```
Comment (gabrielesvelto) [inline apps/system/test/unit/dialer_agent_test.js]: nit: Likewise

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fe8ddca526058c2fe3328b1dfc96cdbfdcf43148

**Comment by alastor0325** [comment]
Thanks Gabriele for reviewing :)

**Pull request merged**